[10.0] [FIX] Solve encoding issue and add AMEX support

sebastienbeau opened the pull request:
This PR solve 3 issues:

- importing AMEX qif bank statement (header is CCARD)
- solving encoding issue by guessing it with the lib "chardet" (optionnal lib)
- make inheritable the way to parse the date

sebastienbeau committed:
[FIX] fix issue with encoding, as encoding may change depending of your provider try to guess it automatically, Also add support of AMEX qif by adding header CCARD, and extract a function to parse the date so you can hack it in a custom module

pedrobaeza commented:
Changing an external dependency in a stable version is not convenient, as it makes all existing installations to break.

sebastienbeau commented:
@pedrobaeza you are right, I can make it optionnal.

sebastienbeau commented:
I have changed my code, now I use utf-8 by default (this should work with the majority of case) if you had the chardet lib, it will try to guess the right encoding.
Also I have refactored the code to be able to inherit the way to parse the date in custom module

pedrobaeza commented on account_bank_statement_import_qif/wizards/account_bank_statement_import_qif.py:
```diff
@@ -12,6 +12,14 @@
+try:
+    import chardet
+except (ImportError, IOError) as err:
+    _logger.debug(err)
```
Put here `chardet = False` instead for easy handling

pedrobaeza commented on account_bank_statement_import_qif/wizards/account_bank_statement_import_qif.py:
```diff
@@ -20,10 +28,26 @@ class AccountBankStatementImport(models.TransientModel):
         return data_file.strip().startswith('!Type:')
 
+    def _get_qif_encoding(self, data_file):
+        try:
```
Don't put the message each time. With the other change, you can do:
```
if chardet:
    chardet.detect(data_file)['encoding']
else:
    return 'utf-8'
```

pedrobaeza reviewed (changes requested):
And please change module version number

sebastienbeau commented:
Thanks for your feedback Pedro !
I have done the fixes
